slash/parallel/worker.py

```diff
@@ -67,8 +67,13 @@ def _stop_keepalive_thread(self):
             self._watchdog_thread.join()
 
     def start_execution(self, app, collected_tests):
-        if os.getpid() != os.getpgid(0):
-            os.setsid()
+        try:
+            if os.getpid() != os.getpgid(0):
```

Comment:
I strongly prefer an initial check for the existence of those in the beginning of the file and then check it here. e.g:
```python
_HAS_PID_PGID = platform.system() != 'Windows'
```